### PR TITLE
Display all parsed elements when parsing emojis in replies

### LINKED ISSUE #4874: Replying to a message with an emoji followed by any text, without any spaces, will cause the text following the emoji to be eaten in the reply
### Checklist

- [X] I'm reporting a problem with Chatterino
- [X] I've verified that I'm running **the most** recent nightly build or stable release
- [X] I've looked for my problem on the [wiki](https://wiki.chatterino.com/Help/)
- [X] I've searched the [issues and pull requests](https://github.com/Chatterino/chatterino2/issues?q=) for similar looking reports

### Describe your issue

```
▪pajaDank
▪ pajaDank test with a space
pajaDank test with no emoji
```

### OS and Chatterino Version

Chatterino Nightly 2.4.6 (commit 75282579) built on 2023-10-06 with Qt 6.5.0, MSVC 193532217 Running on Windows 10 Version 21H2, kernel: 10.0.19044

### PR DESCRIPTION
**Pull request by Nerixyz**
# Description

Previously, only the first parsed element was used. This also adds a helper for Boost's and `std`'s `apply_visitor`/`visit`, which ensures all alternative types from the variants are handled.

Fixes #4874.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,7 @@
 - Bugfix: Fixed too much text being copied when copying chat messages. (#4812, #4830, #4839)
 - Bugfix: Fixed empty page being added when showing out of bounds dialog. (#4849)
 - Bugfix: Fixed issue on Windows preventing the title bar from being dragged in the top left corner. (#4873)
+- Bugfix: Fixed emojis getting lost in replies. (#4875)
```
Comment (Felanbird) [inline CHANGELOG.md:21]:
```suggestion
- Bugfix: Fixed an issue where reply context didn't render correctly if an emoji was touching text. (#4875)
```

**Review by Felanbird — APPROVED** [review]
works on 🅱️ indows, unlucky for the other ones that didn't build

**Comment by Nerixyz** [comment]
Seems like we _do_ need the deduction guidelines even though we're on C++ 20.

**Review by pajlada — CHANGES_REQUESTED** [review]
You can keep the Overloaded type, but since we have another big PR making changes here it'd be best if you just add the loop to fix the issue

**Review by pajlada — APPROVED** [review]

**Pull request merged**